model/lists/ListService.php

```diff
@@ -32,13 +40,81 @@ class ListService extends \tao_models_classes_ListService
     /**
      * Whenever or not a list is editable
      * The Language list should not be editable.
-     * @todo Make two different kind of lists: system list that are not editable and usual list.
-     * @param \core_kernel_classes_Class $listClass
+     *
+     * @param RdfClass $listClass
+     *
      * @return boolean
+     * @todo Make two different kind of lists: system list that are not editable and usual list.
      */
-    public function isEditable(\core_kernel_classes_Class $listClass)
+    public function isEditable(RdfClass $listClass)
     {
         return $listClass->isSubClassOf($this->getClass(TaoOntology::CLASS_URI_LIST))
-        && $listClass->getUri() !== \tao_models_classes_LanguageService::CLASS_URI_LANGUAGES;
+            && $listClass->getUri() !== \tao_models_classes_LanguageService::CLASS_URI_LANGUAGES;
+    }
+
+    public function getListElement(RdfClass $listClass, $uri)
+    {
+        $request = new ValueCollectionSearchRequest();
+        $request->setValueCollectionUri($listClass->getUri())//->setObject()
+        ;
+
+        $result = $this->getValueService()->findAll(
+            new ValueCollectionSearchInput($request)
+        );
+
+        return iterator_to_array($result->getIterator())[0];
+    }
+
+    public function getListElements(RdfClass $listClass, $sort = true, $limit = 0)
+    {
+        $request = new ValueCollectionSearchRequest();
+        $request->setValueCollectionUri($listClass->getUri());
+
+        $result = $this->getValueService()->findAll(
+            new ValueCollectionSearchInput($request)
+        );
+
+        return $result->getIterator();
+    }
+
+    public function removeList(RdfClass $listClass)
+    {
+        //parent::removeList($listClass);
+        throw new \Exception(__METHOD__ . ' is not implemented');
+    }
+
+    public function createListElement(RdfClass $listClass, $label = '')
+    {
+        $newUri = $this->createUri();
+
+        $valueCollection = new ValueCollection(
+            $listClass->getUri(),
+            new Value(null, $newUri, $label)
+        );
+
+        $this->getValueService()->persist($valueCollection);
+    }
+
+    private function getValueService(): ValueCollectionService
+    {
+        return $this->getServiceLocator()->get(ValueCollectionService::class);
+    }
+
+    private function createUri(): string
+    {
+        return $this->getServiceLocator()->get(UriProvider::class)->provide();
+    }
+
+    /**
+     * @param RdfClass $listClass
+     *
+     * @return bool
+     * @throws core_kernel_persistence_Exception
+     */
+    public function isRemote(RdfClass $listClass): bool
+    {
+        $type = $listClass->getOnePropertyValue($listClass->getProperty('http://www.tao.lu/Ontologies/TAO.rdf#ListType'));
+
+        return $type && ($type->getUri() === 'http://www.tao.lu/Ontologies/TAO.rdf#ListRemote');
```

Comment:
Add/use constants for these values:
* `'http://www.tao.lu/Ontologies/TAO.rdf#ListType'`
* `'http://www.tao.lu/Ontologies/TAO.rdf#ListRemote'`